compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -198,6 +198,12 @@ public class BallerinaParserErrorHandler extends AbstractParserErrorHandler {
     private static final ParserRuleContext[] ON_FAIL_OPTIONAL_BINDING_PATTERN =
             { ParserRuleContext.BLOCK_STMT, ParserRuleContext.TYPE_DESC_IN_TYPE_BINDING_PATTERN };
 
+    private static final ParserRuleContext[] GROUPING_KEY_LIST_ELEMENT =
+            { ParserRuleContext.VARIABLE_REF, ParserRuleContext.GROUPING_KEY_LIST_VAR_DECLARATION };
```

Comment:
```suggestion
            { ParserRuleContext.VARIABLE_NAME, ParserRuleContext.GROUPING_KEY_LIST_VAR_DECLARATION };
```
since `variable-reference := identifier | qualified-identifier | xml-qualified-name`